src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2627,19 +2627,13 @@ public static string LastWriteTimeString(PSObject instance)
                             return;
                         }
 
-                        // Junctions cannot have files
-                        if (DirectoryInfoHasChildItems((DirectoryInfo)pathDirInfo))
-                        {
-                            string message = StringUtil.Format(FileSystemProviderStrings.DirectoryNotEmpty, path);
-                            WriteError(new ErrorRecord(new IOException(message), "DirectoryNotEmpty", ErrorCategory.WriteError, path));
-                            return;
-                        }
-
                         if (Force)
                         {
                             try
                             {
                                 pathDirInfo.Delete();
+                                CreateDirectory(path, false);
+                                pathDirInfo = new DirectoryInfo(path);
```

Comment:
Formally we can catch a race condition if `pathDirInfo.Delete()` throws.
I think it is better to move the new two line after the `try-catch` block.